queue/src/fiskaltrust.Middleware.Storage.AzureTableStorage/Repositories/MasterData/AzureTableStoragePosSystemMasterDataRepository.cs

```diff
@@ -15,7 +16,14 @@ public AzureTableStoragePosSystemMasterDataRepository(QueueConfiguration queueCo
 
         public const string TABLE_NAME = nameof(PosSystemMasterData);
 
-        public async Task ClearAsync() => await ClearTableAsync().ConfigureAwait(false);
+        public async Task ClearAsync()
+        {
+            var result = _tableClient.QueryAsync<TableEntity>(select: new List<string>() { "PartitionKey", "RowKey" });
+            await foreach (var item in result)
+            {
+                await _tableClient.DeleteEntityAsync(item.PartitionKey, item.RowKey);
```

Comment:
Do we need the ConfigureAwait stuff here too?

Reply:
puh the original code basically looked like this.

```
        public async Task ClearAsync() => await ClearTableAsync().ConfigureAwait(false);

        protected async Task ClearTableAsync()
        {
            var result = _tableClient.QueryAsync<TableEntity>(select: new List<string>() { "PartitionKey", "RowKey" });
            await foreach (var item in result)
            {
                await _tableClient.DeleteEntityAsync(item.PartitionKey, item.RowKey);
            }
        }
```

Which was now changed to this:

```
        public async Task ClearAsync()
        {
            var result = _tableClient.QueryAsync<TableEntity>(select: new List<string>() { "PartitionKey", "RowKey" });
            await foreach (var item in result)
            {
                await _tableClient.DeleteEntityAsync(item.PartitionKey, item.RowKey);
            }
        }
```

There was no ConfigureAwait on the `DeleteEntityAsync` call of the original method only on the method call. and we've combined that into one method. I think we've preserved the behaviour.

Reply:
Generally speaking, we probably only need the `ConfigureAwait` stuff when we're using packages in Xamarin (at least that's the only place I remember we had deadlocks at without it) - so I guess it's not super crucial here.